modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java

```diff
@@ -0,0 +1,310 @@
+            dependsOnMethods = "testRegisterApplication", dataProvider = "responseModeDataProvider")
+    public void testSendAuthRequestPost(String responseType, String responseModeProvided,
+                                                     String responseMode) throws Exception {
+
+        List<NameValuePair> urlParameters = new ArrayList<>();
+        urlParameters.add(new BasicNameValuePair(OAuth2Constant.OAUTH2_RESPONSE_TYPE, responseType));
+        urlParameters.add(new BasicNameValuePair(OAuth2Constant.OAUTH2_RESPONSE_MODE, responseModeProvided));
+        urlParameters.add(new BasicNameValuePair(OAuth2Constant.OAUTH2_CLIENT_ID, consumerKey));
+        urlParameters.add(new BasicNameValuePair(OAuth2Constant.OAUTH2_REDIRECT_URI, CALLBACK_URL));
+        urlParameters.add(new BasicNameValuePair(OAuth2Constant.OAUTH2_SCOPE, OAuth2Constant.OAUTH2_SCOPE_OPENID_WITH_INTERNAL_LOGIN));
+        urlParameters.add(new BasicNameValuePair(OAuth2Constant.OAUTH2_NONCE, UUID.randomUUID().toString()));
+
+        HttpResponse response = sendPostRequestWithParameters(client, urlParameters,
+                OAuth2Constant.AUTHORIZE_ENDPOINT_URL);
+        Assert.assertNotNull(response, "Authorization request failed. Authorized response is null.");
+
+        String locationValue = getLocationHeaderValue(response);
+        EntityUtils.consume(response.getEntity());
+
+        String sessionDataKeyConsent = DataExtractUtil.getParamFromURIString(locationValue,
+                OAuth2Constant.SESSION_DATA_KEY_CONSENT);
+
+        String sessionDataKey;
+        if (sessionDataKeyConsent == null) {
+            Assert.assertTrue(locationValue.contains(OAuth2Constant.SESSION_DATA_KEY),
+                    "sessionDataKey not found in response.");
+            sessionDataKey = DataExtractUtil.getParamFromURIString(locationValue, OAuth2Constant.SESSION_DATA_KEY);
+            Assert.assertNotNull(sessionDataKey, "sessionDataKey is null.");
+
+            sessionDataKeyConsent = getSessionDataKeyConsent(client, sessionDataKey);
+        }
+
+        response = sendApprovalPost(client, sessionDataKeyConsent);
+        Assert.assertNotNull(response, "Approval request failed. response is invalid.");
+
+        locationValue = getLocationHeaderValue(response);
+
+        AuthorizationCode authorizationCode = getAuthorizationCode(locationValue, responseMode);
+        Assert.assertNotNull(authorizationCode,
+                "Authorization code is null or could not be found.");
+        EntityUtils.consume(response.getEntity());
+    }
+
+    /**
+     * Extract authorization code.
+     * @param url redirection url
+     * @param responseMode response mode which is used to send the request
+     * @return AuthorizationCode object
+     */
+    private AuthorizationCode getAuthorizationCode(String url, String responseMode)
+            throws URISyntaxException {
+        String code;
```

Comment:
```suggestion

        String code;
```

Reply:
fixed with [52a5410](https://github.com/wso2/product-is/pull/16726/commits/52a5410ae119993b886fa486c661dfc5c7cfbe8f)